navigation/image_database.h

```diff
@@ -379,11 +412,82 @@ class ImageDatabase
         return GridRecorder(*this, xrange, yrange, zrange, heading, imageFormat);
     }
 
+    //! Start recording a route
     RouteRecorder getRouteRecorder(const std::string &imageFormat = "png")
     {
         return RouteRecorder(*this, imageFormat);
     }
 
+    //! Get the resolution of saved images
+    cv::Size getResolution() const
+    {
+        BOB_ASSERT(hasMetadata());
+        return m_Resolution;
+    }
+
+    //! Check if this database has any saved metadata (yet)
+    bool hasMetadata() const { return static_cast<bool>(m_MetadataYAML); }
+
+    /**!
+     *  \brief Unwrap all the panoramic images in this database into a new
+     *         folder, creating a new database.
+     */
+    void unwrap(const filesystem::path &destination, const cv::Size &unwrapRes)
+    {
+        // Check that the database doesn't already exist
+        BOB_ASSERT(!(destination / EntriesFilename).exists());
+
+        // Create object for unwrapping images
+        std::string camName;
+        getMetadata()["camera"]["name"] >> camName;
+        ImgProc::OpenCVUnwrap360 unwrapper(getResolution(), unwrapRes, camName);
+        filesystem::create_directory(destination);
+
+        // Copy entries (CSV) file
+        copyfile(m_Path / EntriesFilename, destination / EntriesFilename);
+
+        // Create new metadata (YAML) file
+        {
+            /*
+             * There is no way to edit a persistence file in OpenCV, so we have
+             * to do it ourselves in this ugly way.
+             *
+             * First, copy all fields, except for changing needsUnwrapping to false.
+             */
+            std::string line;
+            std::ofstream ofs((destination / MetadataFilename).str());
+            for (std::ifstream ifs((m_Path / MetadataFilename).str()); std::getline(ifs, line); ) {
+                if (line.find("  needsUnwrapping:") == 0) {
+                    ofs << "  needsUnwrapping: 0\n";
+                } else {
+                    ofs << line << "\n";
```

Comment:
Pretty sure you should always use ``std::endl`` rather than ``"\n"`` with iostreams - I honestly can't quite remember why though

Reply:
I kind of did this by accident, but it seems like it isn't a bad way to do it:
https://en.cppreference.com/w/cpp/io/manip/endl

Apparently ``std::endl`` also flushes the stream every time you call it so it can degrade performance (although that won't matter here)